### ISSUE TITLE
status JSON keeps the downloading files in case of failure.

### ISSUE THREAD

**Issue by zeronone** [SNUArduinoLabs/coe_client#18]
Following is the downloader log:
```
            "2015-08-19 23:33:13,543 [downloader.py     : 201] DEBUG    - Trying to download file [e1100622fd6b48bc46e532305afc82d48b94dc06]\n", 
            "2015-08-19 23:33:14,851 [downloader.py     : 208] INFO     - Finish downloading file [e1100622fd6b48bc46e532305afc82d48b94dc06] (1.31 seconds)\n", 
            "2015-08-19 23:33:14,852 [downloader.py     : 216] DEBUG    - Running checksum check. Using SHA1\n", 
            "2015-08-19 23:33:14,993 [downloader.py     : 239] INFO     - Checksum check PASSED.\n", 
            "2015-08-19 23:33:14,998 [database.py       : 215] DEBUG    - Loading table: file on 
                                                    <Database(sqlite:////home/pi/coe_client/src/db/coe_client.db)>\n", 
            "2015-08-19 23:33:15,145 [downloader.py     : 201] DEBUG    - Trying to download file [ee3bfcdf989ba42e38793c3986305dba66063ab3]\n", 
            "2015-08-19 23:33:15,958 [downloader.py     : 208] INFO     - Finish downloading file [ee3bfcdf989ba42e38793c3986305dba66063ab3] (0.81 seconds)\n", 
            "2015-08-19 23:33:15,959 [downloader.py     : 216] DEBUG    - Running checksum check. Using SHA1\n", 
            "2015-08-19 23:33:15,973 [downloader.py     : 239] INFO     - Checksum check PASSED.\n", 
            "2015-08-19 23:33:16,105 [downloader.py     : 201] DEBUG    - Trying to download file [5ff9ae62d7a4b609262c1965357a054ff7822a3e]\n", 
            "2015-08-19 23:33:16,475 [downloader.py     : 208] INFO     - Finish downloading file [5ff9ae62d7a4b609262c1965357a054ff7822a3e] (0.37 seconds)\n", 
            "2015-08-19 23:33:16,477 [downloader.py     : 216] DEBUG    - Running checksum check. Using SHA1\n", 
            "2015-08-19 23:33:16,480 [downloader.py     : 239] INFO     - Checksum check PASSED.\n", 
            "2015-08-19 23:33:16,617 [downloader.py     : 201] DEBUG    - Trying to download file [885ff3f003f9dcd53b27bb7f6f45507651242dcb]\n", 
            "2015-08-19 23:33:16,698 [downloader.py     : 208] INFO     - Finish downloading file [885ff3f003f9dcd53b27bb7f6f45507651242dcb] (0.08 seconds)\n", 
            "2015-08-19 23:33:16,699 [downloader.py     : 216] DEBUG    - Running checksum check. Using SHA1\n", 
            "2015-08-19 23:33:16,701 [downloader.py     : 239] INFO     - Checksum check PASSED.\n", 
            "2015-08-19 23:33:16,831 [downloader.py     : 201] DEBUG    - Trying to download file [5b84ae6e767df6573065bbafd27e70d1b766b544]\n", 
            "2015-08-19 23:33:17,645 [downloader.py     : 208] INFO     - Finish downloading file [5b84ae6e767df6573065bbafd27e70d1b766b544] (0.81 seconds)\n", 
            "2015-08-19 23:33:17,646 [downloader.py     : 216] DEBUG    - Running checksum check. Using SHA1\n", 
            "2015-08-19 23:33:17,648 [downloader.py     : 239] INFO     - Checksum check PASSED.\n", 
            "2015-08-19 23:33:17,782 [downloader.py     : 201] DEBUG    - Trying to download file [98a267d4610dfc3fcfa470d35c28185cdf37ee7a]\n", 
            "2015-08-19 23:33:18,255 [downloader.py     : 208] INFO     - Finish downloading file [98a267d4610dfc3fcfa470d35c28185cdf37ee7a] (0.47 seconds)\n", 
            "2015-08-19 23:33:18,257 [downloader.py     : 216] DEBUG    - Running checksum check. Using SHA1\n", 
            "2015-08-19 23:33:18,263 [downloader.py     : 239] INFO     - Checksum check PASSED.\n", 
            "2015-08-19 23:33:18,431 [downloader.py     : 173] INFO     - Successfully downloaded ALL files. Sleeping until database is modified...\n", 
```

And the `/status` JSON response still gives 
```
    "downloading": [
        "e1100622fd6b48bc46e532305afc82d48b94dc06", 
        "ee3bfcdf989ba42e38793c3986305dba66063ab3", 
        "5ff9ae62d7a4b609262c1965357a054ff7822a3e", 
        "885ff3f003f9dcd53b27bb7f6f45507651242dcb", 
        "5b84ae6e767df6573065bbafd27e70d1b766b544", 
        "98a267d4610dfc3fcfa470d35c28185cdf37ee7a"
    ], 
```

**Comment by cacyss0807** [comment]
are both logs (`downloader` and `downloading`)from a single `status/` request?

**Comment by zeronone** [comment]
Yes !

**Comment by cacyss0807** [comment]
Then that was an expected output. I changed the way and time I set if a file is being download or not. I will test it. Probably that should fix this (see: 8fe3c42e6ce71cd37a27cb61659c7549e730a767, 66425f4ef08c83e8b9a473e975852e62395baa4b, 3fbb984ae1db4717a5d182898a5c0cb510825d6d)
